src/common/image_cache.c

```diff
@@ -203,7 +203,7 @@ void dt_image_cache_print(dt_image_cache_t *cache)
            (float)cache->cache.cost / (float)cache->cache.cost_quota);
 }
 
-dt_image_t *dt_image_cache_get(dt_image_cache_t *cache, const int32_t imgid, char mode)
+dt_image_t *dt_image_cache_get(dt_image_cache_t *cache, const dt_imgid_t imgid, char mode)
 {
   if(imgid <= 0) return NULL;
```

Comment:
This test is done several times, maybe the macro ?

Reply:
done